crates/pathfinder/examples/generate_tree.rs

```diff
@@ -0,0 +1,175 @@
+use anyhow::Context;
+use stark_hash::{stark_hash, StarkHash};
+use std::cell::RefCell;
+use std::io::BufRead;
+use std::num::NonZeroUsize;
+
+fn main() -> anyhow::Result<()> {
+    let mut args = std::env::args();
+    let myself = args.next().unwrap_or_else(|| String::from("generate_tree"));
+    let any = args.next().is_some();
+
+    if any {
+        eprintln!("USAGE: {myself}");
+        eprintln!();
+        eprintln!("Reads stdin for whitespace separated tuples or triples of 0x prefixed starkhashes, where key is the first value.");
+        eprintln!("Outputs the starknet merkle patricia tree for the input without leaf nodes.");
+        eprintln!("(Leaf nodes used to be part before pathfinder v0.3.0.)");
+        eprintln!();
+        eprintln!("Usage example:
+./{myself} << EOF
+0x84 0x3
+EOF
+
+Above generates the py/test_call.py::populate_test_contract_with_132_on_3 tree_contracts table.
+
+./{myself} << EOF
+0x57dde83c18c0efe7123c36a52d704cf27d5c38cdf0b1e1edc3b0dae3ee4e374 0x050b2148c0d782914e0b12a1a32abe5e398930b7e914f82c65cb7afce0a0ab9b 0x04fb440e8ca9b74fc12a22ebffe0bc0658206337897226117b985434c239c028
+EOF
+
+Above generates the py/test_call.py::populate_test_contract_with_132_on_3 tree_global table.");
+
+        // it would be nice if you could invoke tree of another path, like
+        // 0x1 class_hash $(contract-0x1.tree)
+        // but that is quite annoying to do right now because merkletree takes ownership, cannot
+        // how many folds are being done for the values..
```

Comment:
incomplete comment?

Reply:
Not really. I was thinking for a single toml description one could use for the whole state.... But couldn't really do it. Hmm... Looks like toml would support quoted keys and looks like unquoted the `0x` hex would work just as well. That might actually work to describe block or multiple blocks even in a single file.